packages/hdwallet-gridplus/src/bitcoin.ts

```diff
@@ -1,711 +1,139 @@
+          },
         });
 
-        if (!pubkey || !pubkey.length) {
-          throw new Error(`No public key for output`);
-        }
-
-        const pubkeyBuffer = Buffer.isBuffer(pubkey[0]) ? pubkey[0] : Buffer.from(pubkey[0], "hex");
+        if (!sig) throw new Error(`No signature returned from device for input ${i}`);
 
-        const pubkeyHex =
-          pubkeyBuffer.length === 65
-            ? Buffer.from(pointCompress(pubkeyBuffer, true)).toString("hex")
-            : pubkeyBuffer.toString("hex");
+        const { r, s } = sig;
 
-        const scriptType =
-          (output.scriptType as unknown as core.BTCInputScriptType) || core.BTCInputScriptType.SpendAddress;
-        address = deriveAddressFromPubkey(pubkeyHex, msg.coin, scriptType);
-      } else {
-        throw new Error("Output must have either address or addressNList");
-      }
-
-      const { scriptPubKey: finalScriptPubKey } = (() => {
-        // Native SegWit (bech32): ltc1 for Litecoin, bc1 for Bitcoin
-        if (address.startsWith("ltc1") || address.startsWith("bc1")) {
-          const decoded = bech32.decode(address);
-          const hash160 = Buffer.from(bech32.fromWords(decoded.words.slice(1)));
-
-          const scriptPubKey = bitcoin.script.compile([bitcoin.opcodes.OP_0, hash160]);
-          return { scriptPubKey };
-        }
-
-        // Bitcoin Cash CashAddr format: bitcoincash: prefix or q for mainnet
-        if (address.startsWith("bitcoincash:") || address.startsWith("q")) {
-          const legacyAddress = bchAddr.toLegacyAddress(address);
-          const decoded = bs58Decode(legacyAddress);
-          const versionByte = decoded[0];
-          const hash160 = decoded.slice(1);
+        if (!Buffer.isBuffer(r)) throw new Error("Invalid signature (r)");
+        if (!Buffer.isBuffer(s)) throw new Error("Invalid signature (s)");
 
-          // Check if P2SH (Bitcoin Cash uses 0x05 for P2SH)
-          if (versionByte === network.scriptHash) {
-            const scriptPubKey = bitcoin.script.compile([
-              bitcoin.opcodes.OP_HASH160,
-              hash160,
-              bitcoin.opcodes.OP_EQUAL,
-            ]);
-            return { scriptPubKey };
-          }
-
-          // P2PKH
-          const scriptPubKey = bitcoin.script.compile([
-            bitcoin.opcodes.OP_DUP,
-            bitcoin.opcodes.OP_HASH160,
-            hash160,
-            bitcoin.opcodes.OP_EQUALVERIFY,
-            bitcoin.opcodes.OP_CHECKSIG,
-          ]);
-          return { scriptPubKey };
-        }
+        return Buffer.concat([r, s]);
+      },
+    };
 
-        // Other Base58 addresses (P2PKH or P2SH)
-        const decoded = bs58Decode(address);
-        const versionByte = decoded[0];
-        const hash160 = decoded.slice(1);
+    // single sha256 hash and allow device to perform second sha256 hash when signing (pending gridplus fix for hashType.NONE)
+    await psbt.signInputAsync(i, signer, undefined, true);
```

Comment:
Seeing this guy:

<img width="1106" height="80" alt="Image" src="https://github.com/user-attachments/assets/fd45e1e1-6ff6-429c-9cbd-3efd5d20efab" />

Also, I get `yarn.lock` generated here when running `yarn`, which could be the same root cause

Is `yarn && yarn build` working for you locally @kaladinlight , without regenerated yarn.lock?